pkg/virt-controller/watch/application.go

```diff
@@ -69,7 +76,21 @@ func Execute() {
 
 	app.restClient = app.clientSet.RestClient()
 
-	restful.Add(rest.WebService)
+	readinessFunc := func(_ *restful.Request, response *restful.Response) {
+		res := map[string]interface{}{}
+		if !app.isLeader {
+			res["apiserver"] = map[string]interface{}{"leader": "failed", "error": "current pod is not leader"}
+			response.WriteHeaderAndJson(http.StatusInternalServerError, res, restful.MIME_JSON)
```

Comment:
I think 503 would be a better return value. It is not an internal server error, if we did not yet win an election.